src/containerapp/azext_containerapp/tests/latest/test_containerapp_env_commands.py

```diff
@@ -15,6 +15,102 @@
+
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+        self.cmd('containerapp env identity show -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('type', 'SystemAssigned'),
+        ])
+
+        self.cmd('containerapp env identity remove --system-assigned -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('type', 'None'),
+        ])
+
+        self.cmd('containerapp env identity assign --system-assigned -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('type', 'SystemAssigned'),
+        ])
+
+        self.cmd('containerapp env identity remove --system-assigned -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('type', 'None'),
+        ])
+
+        self.cmd('containerapp env show -n {} -g {}'.format(env_name, resource_group), checks=[
+            JMESPathCheck('name', env_name),
+        ])
+
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer(location="northeurope")
+    def test_containerapp_env_identity_user(self, resource_group):
+        # MSI is not available in North Central US (Stage), if the TEST_LOCATION is "northcentralusstage", use eastus as location
+        location = TEST_LOCATION
+        if format_location(location) == format_location(STAGE_LOCATION):
+            location = "eastus2euap"
+        self.cmd('configure --defaults location={}'.format(location))
+
+        user_identity_name1 = self.create_random_name(prefix='env-msi1', length=24)
+        user_identity_name2 = self.create_random_name(prefix='env-msi2', length=24)
+        user_identity_id1 = self.cmd('identity create -g {} -n {}'.format(resource_group, user_identity_name1)).get_output_in_json()["id"]
+        user_identity_id2 = self.cmd('identity create -g {} -n {}'.format(resource_group, user_identity_name2)).get_output_in_json()["id"]
+
+        env_name = self.create_random_name(prefix='containerapp-e2e-env', length=24)
+        self.cmd('containerapp env create -g {} -n {} --mi-user-assigned {} {} --logs-destination none'.format(resource_group, env_name, user_identity_id1, user_identity_id2))
+
+        containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+        while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+        self.cmd('containerapp env identity show -g {} -n {}'.format(resource_group, env_name), checks=[
+            JMESPathCheck('type', 'UserAssigned'),
```

Comment:
Please check `user_identity_id1` and `user_identity_id2` at the show command if it should return.

Reply:
As the user_identity_id2 with become key, you can check as following:
JMESPathCheck(f'userAssignedIdentities.{user_identity_id1}', {}),